### DISCUSSION TITLE
Why extra API call?

### DISCUSSION THREAD

**Discussion by ajmalab** [hub4j/github-api#1845] [Q&A] [answered]
Hello all,

Been testing out this library and noticed that we require more API calls than necessary with the approach taken in this library. For eg:, if I'd like to post to the `dispatch` endpoint, I'd have to do `githubClient.getRepository("my-repo").dispatch()`. The `getRepository()` makes an additional GET call, instead of simply constructing the URL to be `/repos/owner/repo/dispatches`. 

Curious, why was this approach taken over simply constructing the required endpoint (considering the restrictive API limits for Github)? Is there any caching being done to avoid multiple getRepository calls for the same repository? Or is the consuming service expected to handle the caching? Perhaps it's somewhere in the docs, but I could not find it.

**Comment by bitwiseman** [comment]
Not sure:
https://github.com/hub4j/github-api/blob/2e6dcc1137189aefb9c4c12b7d15b7a0b54de6d1/src/main/java/org/kohsuke/github/GitHub.java#L682-L687

Not sure how this is working in your code given the above...

**Reply by ajmalab** [reply]
Hi, not sure I understand what you mean. If I understand that code correctly, getRepository() makes an API call?

**Comment by bitwiseman** [comment]
`GHRepository.read(this, tokens[0], tokens[1]); `  = one api call.

**Reply by ajmalab** [reply]
yes, but to do a `dispatch()` call, I have to do `GitHub.getRepository().dispatch()`, which is 2 API calls. I do not use the response from `getRepository()` at all, so could save a round-trip if I can avoid this. At least this is my understanding, could not find a way to call `dispatch()` without calling `getRepository()`.

**Reply by bitwiseman** [reply, marked as answer]
Yes, that is an potential enhancement. Please feel free to file an issue for this.
